cmd/create/tuningconfigs/cmd.go

```diff
@@ -0,0 +1,133 @@
+limitations under the License.
+*/
+
+package tuningconfigs
+
+import (
+	"fmt"
+	"os"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift/rosa/pkg/input"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+var args struct {
+	name     string
+	specPath string
+}
+
+var Cmd = &cobra.Command{
+	Use:     "tuning-configs",
+	Aliases: []string{"tuningconfig", "tuningconfigs", "tuning-config"},
+	Short:   "Add tuning config",
+	Long:    "Add a tuning config to a cluster.",
+	Example: `  # Add a tuning config with name "tuned1" and spec from a file "file1" to a cluster named "mycluster"
+ rosa create tuning-config --name=tuned1 --spec-path=file1 --cluster=mycluster"`,
+	Run: run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	ocm.AddClusterFlag(Cmd)
+
+	flags.StringVar(
+		&args.name,
+		"name",
+		"",
+		"Name of the tuning config to add.",
+	)
+	flags.StringVar(
+		&args.specPath,
+		"spec-path",
+		"",
+		"Path of the file containing the spec section of the tuning config to add.",
+	)
+
+	interactive.AddFlag(flags)
+}
+
+func run(cmd *cobra.Command, _ []string) {
+	r := rosa.NewRuntime().WithAWS().WithOCM()
+	defer r.Cleanup()
+
+	clusterKey := r.GetClusterKey()
+	cluster := r.FetchCluster()
+
+	var err error
+	name := args.name
+	if interactive.Enabled() {
+		name, err = interactive.GetString(interactive.Input{
+			Question: "Name of the tuning config",
+			Help:     cmd.Flags().Lookup("name").Usage,
+			Default:  name,
+			Required: true,
+		})
+		if err != nil {
+			r.Reporter.Errorf("Expected a valid name: %s", err)
+			os.Exit(1)
+		}
+	}
+
+	specPath := args.specPath
+	if interactive.Enabled() {
+		specPath, err = interactive.GetString(interactive.Input{
+			Question: "Path of the file containing the spec of the tuning config",
+			Help:     cmd.Flags().Lookup("spec-path").Usage,
+			Default:  specPath,
+			Required: true,
+		})
+		if err != nil {
+			r.Reporter.Errorf("Expected a valid spec path: %v", err)
+			os.Exit(1)
+		}
+	}
+
+	tuningConfig, err := buildTuningConfigFromInputFile(specPath, name, clusterKey)
+	if err != nil {
+		r.Reporter.Errorf("%v", err)
+		os.Exit(1)
+	}
+
+	_, err = r.OCMClient.CreateTuningConfig(cluster.ID(), tuningConfig)
+	if err != nil {
+		r.Reporter.Errorf("Failed to add tuning config to cluster '%s': %v", clusterKey, err)
+		os.Exit(1)
+	}
+
+	r.Reporter.Infof("Tuning config '%s' has been created on cluster '%s'.", name, clusterKey)
+	r.Reporter.Infof("To view all tuning configs, run 'rosa list tuning-configs -c %s'", clusterKey)
+}
+
+func buildTuningConfigFromInputFile(specPath string, name string, clusterKey string) (*cmv1.TuningConfig, error) {
+	specJson, err := input.UnmarshalInputFile(specPath)
+	if err != nil {
+		return nil, fmt.Errorf("Expected a valid TuneD spec file: %v", err)
+	}
+
+	tuningConfigBuilder := cmv1.NewTuningConfig().Name(name).Spec(specJson)
```

Comment:
Does that also work with yaml? Should there be a warning for the user or validation that it is a json file?

Reply:
Json only. Validation is done on backend side, you will get an error like.

```
rosa create tuningconfig -c ad2 --name tuned-2 --spec-path invalidspec.json
E: Expected a valid TuneD spec file: invalid character 'i' looking for beginning of value
```